sql/moz-fx-data-shared-prod/subscription_platform/stripe_subscriptions/view.sql

```diff
@@ -1,121 +1,151 @@
 CREATE OR REPLACE VIEW
   `moz-fx-data-shared-prod.subscription_platform.stripe_subscriptions`
 AS
-WITH stripe_subscriptions AS (
+WITH subscription AS (
   SELECT
-    customer AS customer_id,
+    customer_id,
     id AS subscription_id,
-    plan AS plan_id,
     status,
-    event_timestamp,
-    MIN(COALESCE(trial_end, start_date)) OVER (
-      PARTITION BY
-        customer
-      ROWS BETWEEN
-        UNBOUNDED PRECEDING
-        AND UNBOUNDED FOLLOWING
-    ) AS customer_start_date,
+    _fivetran_synced,
+    -- TODO verify logic for subscription_start_date
     COALESCE(trial_end, start_date) AS subscription_start_date,
     created,
+    -- trial_start,
     trial_end,
+    -- current_period_end,
+    -- current_period_start,
     canceled_at,
-    mozfun.map.get_key(metadata, "cancelled_for_customer_at") AS canceled_for_customer_at,
+    JSON_VALUE(metadata, "$.cancelled_for_customer_at") AS canceled_for_customer_at,
     cancel_at,
     cancel_at_period_end,
-    ended_at,
-    discount.promotion_code AS promotion_code_id,
+    IF(ended_at <= _fivetran_synced, ended_at, NULL) AS ended_at,
   FROM
-    mozdata.stripe.subscriptions
+    `moz-fx-data-bq-fivetran`.stripe.subscription_history
   WHERE
     status NOT IN ("incomplete", "incomplete_expired")
+    -- choose subscription from subscription history
+    AND _fivetran_active
 ),
-stripe_customers AS (
+subscription_item AS (
+  SELECT
+    id AS subscription_item_id,
+    subscription_id,
+    plan_id,
+  FROM
+    `moz-fx-data-bq-fivetran`.stripe.subscription_item
+),
+customer AS (
   SELECT
     id AS customer_id,
-    mozfun.map.get_key(metadata, "fxa_uid") AS fxa_uid,
+    TO_HEX(SHA256(JSON_VALUE(metadata, "$.userid"))) AS fxa_uid,
+    address_country,
   FROM
-    mozdata.stripe.customers
+    `moz-fx-data-bq-fivetran`.stripe.customer
+  UNION ALL
+  -- include customer records that were deleted before the initial fivetran stripe import
+  SELECT
+    id AS customer_id,
+    JSON_VALUE(metadata, "$.fxa_uid") AS fxa_uid,
```

Comment:
The difference between `TO_HEX(SHA256(JSON_VALUE(metadata, "$.userid"))) AS fxa_uid,` and `JSON_VALUE(metadata, "$.fxa_uid") AS fxa_uid,` on either side of this `UNION ALL` is because of https://github.com/mozilla/bigquery-etl/blob/78885a77dd17bc26585e2f4ad7d0c2199da02ee2/bigquery_etl/stripe/allowed_fields.py#L113-L117